### PR TITLE
fix: selectable sidebar menu items - WPB-9727

### PR DESCRIPTION
**Pull request by caldrian**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-9727" title="WPB-9727" target="_blank">TaskWPB-9727</a>  Implement UISplitViewController for iPad
  </td></table>
  <br />
 

### Issue

Currently in the sidebar only the conversation filter menu items are selectable. The other menu had simple tap handlers.

This PR makes the sidebar behave similar to a tab controller:
The conversation filters, the archive, the connect item and the settings, all menu items will change the view in the middle column of the split view controller. (see the attached gif)

Also the code for the previews in Xcode (and thus also for the snapshot tests) are moved into separate files.

### Testing

Can be tested either in the Xcode preview area or after the navigation overhaul epic is merged.

---

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [x] Adds/updates automated tests.

---

### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_
- [x] Make sure you use the API for UI elements that support large fonts.
- [x] All colors are taken from WireDesign.ColorTheme or constructed using WireDesign.BaseColorPalette.
- [x] New UI elements have Accessibility strings for VoiceOver.

ezgif-6-a28c2e2b8d

### CODE REVIEW

File: WireUI/Tests/WireSidebarTests/Resources/ReferenceImages/SidebarMenuItemSnapshotTests/testColorSchemeVariants.dark.png
Comment (caldrian) [inline WireUI/Tests/WireSidebarTests/Resources/ReferenceImages/SidebarMenuItemSnapshotTests/testColorSchemeVariants.dark.png:1]: Displaying indicator to be able to verify that the icon widths are the same for all items in a menu.

File: WireUI/Sources/WireSidebar/Views/SidebarViewController.swift [hunk 1]
```diff
@@ -19,94 +19,113 @@
 import SwiftUI
 import WireFoundation
 
-public final class SidebarViewController: UIHostingController<SidebarViewAdapter<AnyView>> {
```
Comment (samwyndham) [inline WireUI/Sources/WireSidebar/Views/SidebarViewController.swift:22]: Any specific reason to change to a regular `UIViewController` subclass? I've nothing against it, just wondering why the change?

**Review by samwyndham — APPROVED** [review]
Looking good!

File: WireUI/Sources/WireSidebar/Views/SidebarViewController.swift (hunk 1 above)
Comment (caldrian) [reply WireUI/Sources/WireSidebar/Views/SidebarViewController.swift:22]: Yes, because I need the view controller to be public, however, I prefer not to expose the type `SidebarViewAdapter`.  Thats' why I chose to add the hosting controller as a child instead.

File: WireUI/Sources/WireSidebar/Views/Preview/SidebarMenuItemPreview.swift [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    var body: some View {
+        VStack {
+            // Display two separate menus here in order to verify, that the size
+            // (especially the width) of the icons is equal for all items within their menu only.
+            SidebarMenuItemContainer(iconSize: $topMenuIconSize) { iconSize in
+                SidebarMenuItemView(icon: "text.bubble", iconSize: iconSize, isHighlighted: false, title: { Text("Regular") }, action: { print("show all conversations") })
+                SidebarMenuItemView(icon: "gamecontroller", iconSize: iconSize, isHighlighted: true, title: { Text("Initially highlighted") }, action: { print("show all conversations") })
+                SidebarMenuItemView(icon: "person.3", iconSize: iconSize, isLink: true, title: { Text("Initially highlighted") }, action: { print("show all conversations") })
+            }
+            .overlay {
```
Comment (netbe) [inline WireUI/Sources/WireSidebar/Views/Preview/SidebarMenuItemPreview.swift:35]: nit-pick: especially in switchUI views, I prefer having each param returning to the next line. Just sharing a thought about formatting, it will be easier to fix when swift format is in place (aka don't fix all your views yet)

```suggestion
                SidebarMenuItemView(icon: "text.bubble", 
                                                       iconSize: iconSize,
                                                       isHighlighted: false,
                                                       title: { Text("Regular") }) { 
                                                          print("show all conversations") 
                                                       }
     ```

File: WireUI/Sources/WireSidebar/Views/Preview/SidebarPreview.swift [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import SwiftUI
+import WireFoundation
+
+@available(iOS 17.0, *)
```
Comment (netbe) [inline WireUI/Sources/WireSidebar/Views/Preview/SidebarPreview.swift:22]: why iOS 17 is there an alternate component for previous version?

File: WireUI/Sources/WireSidebar/Views/Preview/SidebarPreview.swift [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    var body: some View {
+        NavigationSplitView(
+            columnVisibility: $columnVisibility,
+            preferredCompactColumn: $preferredCompactColumn,
+            sidebar: {
+                SidebarView(
+                    accountInfo: accountInfo,
+                    selectedMenuItem: $selectedMenuItem,
+                    accountImageAction: {},
+                    supportAction: {},
```
Comment (netbe) [inline WireUI/Sources/WireSidebar/Views/Preview/SidebarPreview.swift:45]: can that be removed if that's default actions?

File: WireUI/Sources/WireSidebar/Views/Preview/SidebarPreview.swift [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+                    accountImageView: { uiImage, availability in
+                        MockAccountImageView(uiImage: uiImage, availability: availability)
+                    }
+                )
+                .navigationSplitViewColumnWidth(260)
```
Comment (netbe) [inline WireUI/Sources/WireSidebar/Views/Preview/SidebarPreview.swift:50]: is that number linked to SidebarViewControllerPreview constant too? in that case consider using a constant

File: WireUI/Sources/WireSidebar/Views/SidebarView.swift [hunk 6]
```diff
@@ -90,58 +97,50 @@ public struct SidebarView<AccountImageView>: View where AccountImageView: View {
...
-                .padding(.top, 12)
-            SidebarMenuItem(
-                icon: "person.badge.plus",
-                iconSize: iconSize
-            ) {
-                Text("sidebar.contacts.connect.title", bundle: .module)
-            } action: {
-                connectAction()
-            }
+            menuItemHeader("sidebar.contacts.title")
```
Comment (netbe) [inline WireUI/Sources/WireSidebar/Views/SidebarView.swift:119]: question: we don't use SwiftGen localizable constants here?

File: WireUI/Sources/WireSidebar/Views/Preview/SidebarPreview.swift (hunk 3 above)
Comment (netbe) [reply WireUI/Sources/WireSidebar/Views/Preview/SidebarPreview.swift:22]: so that's only for previews right?

**Review by netbe — APPROVED** [review]
LGTM;) just a few comments

File: WireUI/Sources/WireSidebar/Views/SidebarView.swift (hunk 6 above)
Comment (caldrian) [reply WireUI/Sources/WireSidebar/Views/SidebarView.swift:119]: I couldn't get it to work with Swift packages. Then I came to an agreement with my PR reviewer that I use String catalogs, which give visual feedback if something is wrong (like unused keys in the catalog).
Moreover I've added the launch argument
`-NSShowNonLocalizedStrings YES`
which not only displays a warning to the console but also shows all unlocalized strings in capital letters.
Screenshot 2024-09-24 at 09 24 08

File: WireUI/Sources/WireSidebar/Views/Preview/SidebarMenuItemPreview.swift (hunk 2 above)
Comment (caldrian) [reply WireUI/Sources/WireSidebar/Views/Preview/SidebarMenuItemPreview.swift:35]: Fine for me. I changed it for this file and will keep it in mind.

File: WireUI/Sources/WireSidebar/Views/Preview/SidebarPreview.swift (hunk 3 above)
Comment (caldrian) [reply WireUI/Sources/WireSidebar/Views/Preview/SidebarPreview.swift:22]: Yes, this is only for previewing the view. 
The preview uses `NavigationSplitView` which is available from iOS 17 only.
In the project we currently use the `UISplitViewController`.

File: WireUI/Sources/WireSidebar/Views/Preview/SidebarPreview.swift (hunk 5 above)
Comment (caldrian) [reply WireUI/Sources/WireSidebar/Views/Preview/SidebarPreview.swift:50]: done

File: WireUI/Sources/WireSidebar/Views/Preview/SidebarPreview.swift (hunk 4 above)
Comment (caldrian) [reply WireUI/Sources/WireSidebar/Views/Preview/SidebarPreview.swift:45]: This code is for preview, where I didn't implement button handlers. I hope you don't mind.

File: WireUI/Sources/WireSidebar/Views/Preview/SidebarMenuItemPreview.swift (hunk 2 above)
Comment (caldrian) [reply WireUI/Sources/WireSidebar/Views/Preview/SidebarMenuItemPreview.swift:35]: Unfortunately I changed it on the wrong branch, now the changes are visible in this PR: https://github.com/wireapp/wire-ios/pull/1934/files
